### PR TITLE
Download Certain Files from SRA Using Aspera

### PR DESCRIPTION
**Pull request by Miserlou**
This adds support for Aspera rather FTP when downloading from certain SRA URLs. This results in an _extremely significant_ download speed improvement.

A test is included. This test _may fail_ on the UPenn network because of how Aspera works, but it works on Circle and from EC2.

Also included in this PR is the ability to add the word "noslow" into a commit message to avoid running the super long test during a build.

### CODE REVIEW

File: workers/data_refinery_workers/downloaders/test_sra.py [hunk 1]
```diff
@@ -199,3 +197,34 @@ def raise_exception(job_dir):
...
+            experiment_title="It doesn't really matter.",
+            organism_id=9031,
+            organism_name="GALLUS GALLUS",
+            release_date="2013-07-19",
+            last_uploaded_date="2017-09-11",
+            status=BatchStatuses.NEW.value
+        )
+        batch.save()
+
+        # This is converted to us Aspera
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/downloaders/test_sra.py]: What does this comment mean?

File: workers/data_refinery_workers/downloaders/test_sra.py [hunk 2]
```diff
@@ -199,3 +197,34 @@ def raise_exception(job_dir):
...
+        file = File(size_in_bytes=0,
+                    download_url="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR036/ERR036000/ERR036000_1.fastq.gz",  # noqa
+                    raw_format="fastq.gz",
+                    processed_format="tar.gz",
+                    name="ERR036000_1.fastq.gz",
+                    internal_location="IlluminaHiSeq2000/SALMON",
+                    batch=batch)
+        # This is converted to us Aspera
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/downloaders/test_sra.py]: Same question.

File: workers/data_refinery_workers/downloaders/test_sra.py (hunk 1 above)
Comment (kurtwheeler) [reply workers/data_refinery_workers/downloaders/test_sra.py]: Oh maybe "to use Aspera"?

File: workers/data_refinery_workers/downloaders/sra.py [hunk 3]
```diff
@@ -13,9 +14,25 @@
 CHUNK_SIZE = 1024 * 256
 
 
-def _download_file(file: File, downloader_job: DownloaderJob, target_file_path: str) -> bool:
+def _download_file(file: File, downloader_job: DownloaderJob, target_file_path: str, force_ftp: bool=False) -> bool:
+    """ Download file dispatcher. Dispatches to the FTP or Aspera downloader """
+
+    # SRA files have Apsera downloads.
+    if 'ftp.sra.ebi.ac.uk' in file.download_url and not force_ftp:
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/downloaders/sra.py]: Are there files we download that do not have this in their URL?

Comment (Miserlou) [reply workers/data_refinery_workers/downloaders/sra.py]: Yes, I think so. At least from the tests.

File: workers/run_individual_test.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+docker run \
+       --add-host=database:$DB_HOST_IP \
+       --add-host=nomad:$NOMAD_HOST_IP \
+       --env-file workers/environments/test \
+       --volume $volume_directory:/home/user/data_store \
+       --link drdb:postgres \
+       --link nomad:nomad \
+       -i dr_worker_tests python3 manage.py test "$1" --no-input
+      # ex: data_refinery_workers.processors.test_salmon.SalmonTestCase.test_success
+      # ex: data_refinery_workers.downloaders.test_sra.DownloadSraTestCase.test_aspera_downloader
```
[the remark this answers is not in the document]
Comment (Miserlou) [reply workers/run_individual_test.sh]: workers/run_tests.sh isn't capable running an individual test without modification

[the remark this answers is not in the document]
Comment (Miserlou) [reply workers/run_individual_test.sh]: Removed

File: workers/data_refinery_workers/downloaders/test_sra.py (hunk 2 above)
Comment (Miserlou) [reply workers/data_refinery_workers/downloaders/test_sra.py]: Removed

File: workers/data_refinery_workers/downloaders/test_sra.py (hunk 1 above)
Comment (Miserlou) [reply workers/data_refinery_workers/downloaders/test_sra.py]: Clarified

File: workers/Dockerfile [hunk 5]
```diff
@@ -13,5 +13,9 @@ RUN pip3 install common/$(ls common -1 | sort --version-sort | tail -1)
 COPY workers/ .
 
 USER user
+# Install Aspera
+ADD --chown=user:user https://download.asperasoft.com/download/sw/cli/3.7.7/aspera-cli-3.7.7.608.927cce8-linux-64-release.sh aspera-cli-3.7.7.608.927cce8-linux-64-release.sh
+RUN sh aspera-cli-3.7.7.608.927cce8-linux-64-release.sh
+RUN rm aspera-cli-3.7.7.608.927cce8-linux-64-release.sh
```
[the remark this answers is not in the document]
Comment (Miserlou) [reply workers/Dockerfile]: This is the only one I'm leaving the same for now as there is no documentation for how `ccdl/data_refinery_worker_base` was built and published, so I think refactoring/documenting that is outside the scope of this PR.

[the remark this answers is not in the document]
Comment (kurtwheeler) [reply workers/Dockerfile]: :+1:

**Pull request merged**